File: packages/var/tests/parse.test.ts
import { expect, test } from 'vitest'
import { parse } from '../src/parse.js'

test('parse returns a Bdd whose Examples come from paragraphs and carry the heading stack', () => {
  const source = '# Hello\n\nbody'
  const bdd = parse('hello.bdd.md', source)
  expect(bdd.path).toBe('hello.bdd.md')
  expect(bdd.source).toBe(source)
  // One paragraph, one Example. Example name is computed by the planner, not
  // captured here; the structurer's job is just to track scope + body.
  expect(bdd.examples).toHaveLength(1)
  expect(bdd.examples[0]?.scopeStack).toEqual(['Hello'])
})

File: java/var-core/src/test/java/dev/varar/core/ParseTest.java
package com.oselvar.var.core;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.oselvar.var.core.Ast.VarDoc;
import java.util.List;
import org.junit.jupiter.api.Test;

/** Port of {@code typescript/packages/var-core/tests/parse.test.ts}. */
class ParseTest {

    @Test
    void parseReturnsAVarDocWhoseExamplesComeFromParagraphsAndCarryTheHeadingStack() {
        String source = "# Hello\n\nbody";
        VarDoc varDoc = Parse.parse("hello.md", source);
        assertEquals("hello.md", varDoc.path());
        assertEquals(source, varDoc.source());
        // One paragraph, one Example. Example name is computed by the planner, not captured
        // here; the structurer's job is just to track scope + body.
        assertEquals(1, varDoc.examples().size());
        assertEquals(List.of("Hello"), varDoc.examples().get(0).scopeStack());
    }
}
